### ISSUE TITLE
Improve error handling/messaging

### ISSUE THREAD

**Issue by rmcvey** [Netflix-Skunkworks/stethoscope-app#122]
* **I'm submitting a ...**
  - [ ] bug report
  - [x] feature request
  - [ ] support request => Please do not submit support request here, see note at the top of this template.

* **What is the current behavior?**

Pretty much the only error users ever see is "Request timeout", which isn't helpful to anyone. We need to do a better job of surfacing actual errors to end users, even if only to help debug issues that pop up.

* **What is the expected behavior?**

tldr; errors aren't swallowed by app. 

* **What is the motivation / use case for changing the behavior?**

Visibility for the user when things go wrong, ultimately this data will be helpful when we need to resolve issue.

* **Please tell us about your environment:**

  - Stethoscope Version:
  - Platform:
    - [x] Mac
    - [x] Windows
    - [x] Linux

**Comment by rmcvey** [comment]
Fixed in v3.0.5
